src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php

```diff
@@ -102,7 +102,7 @@ private function stripTypeHints($tokens)
                     if ($this->tokenHasType($token, T_STRING) && !$this->currentClass) {
                         $this->currentClass = $token[1];
                     }
-                    elseif($this->tokenHasType($token, T_FUNCTION)) {
+                    elseif($this->tokenHasType($token, T_FUNCTION) && $this->functionShouldBeRead($tokens, $index)) {
```

Comment:
would it make more sense, instead of having this exception called functionShouldBeRead() to introduce a new state e.g. STATE_READING_EXAMPLE which gets ignored? that way it would be clear that this exception is about typehints in spec examples.